rma_analytic/tests/test_rma_analytic.py

```diff
@@ -0,0 +1,174 @@
+        )
+        cls.operation_1 = cls.rma_op_obj.create(
+            {
+                "code": "TEST",
+                "name": "Refund and receive",
+                "type": "customer",
+                "receipt_policy": "ordered",
+                "refund_policy": "ordered",
+                "in_route_id": cls.rma_route_cust.id,
+                "out_route_id": cls.rma_route_cust.id,
+            }
+        )
+        cls.product_1.update(
+            {
+                "rma_customer_operation_id": cls.operation_1.id,
+                "rma_supplier_operation_id": cls.cust_refund_op.id,
+            }
+        )
+        products2move = [
+            (cls.product_1, 3),
+            (cls.product_2, 5),
+            (cls.product_3, 2),
+        ]
+        cls.rma_ana_id = cls._create_rma_from_move(
+            products2move,
+            "supplier",
+            cls.env.ref("base.res_partner_2"),
+            dropship=False,
+        )
+        cls.company_id = cls.env.user.company_id
+        cls.anal = cls.env["account.analytic.account"].create({"name": "Name"})
+        cls.inv_customer = cls.env["account.move"].create(
+            {
+                "partner_id": cls.partner_id.id,
+                "move_type": "out_invoice",
+                "invoice_date": fields.Date.from_string("2023-01-01"),
+                "currency_id": cls.company_id.currency_id,
+                "invoice_line_ids": [
+                    (
+                        0,
+                        None,
+                        {
+                            "name": cls.partner_id.name,
+                            "product_id": cls.product_1.id,
+                            "product_uom_id": cls.product_1.uom_id.id,
+                            "quantity": 12.0,
+                            "price_unit": 100.0,
+                            "analytic_account_id": cls.anal.id,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    @classmethod
+    def _prepare_move(cls, product, qty, src, dest, picking_in):
+        res = super(TestRmaAnalytic, cls)._prepare_move(
+            product, qty, src, dest, picking_in
+        )
+        analytic_1 = cls.env["account.analytic.account"].create(
+            {"name": "Test account #1"}
+        )
+        res.update({"analytic_account_id": analytic_1.id})
+        return res
+
+    def test_analytic(self):
+        for line in self.rma_ana_id.rma_line_ids:
+            for move in line.move_ids:
+                self.assertEqual(
+                    line.analytic_account_id,
+                    move.analytic_account_id,
+                    "the analytic account is not propagated",
+                )
+
+    def test_invoice_analytic(self):
+        """Test wizard to create RMA from a customer invoice."""
+        rma_line = (
+            self.env["rma.order.line"]
+            .with_context(customer=True)
+            .new(
```

Comment:
Could you use [the Form class](https://github.com/ForgeFlow/stock-rma/blob/16.0/rma_account/tests/test_rma_account.py#L236) instead of calling the onchange dfirectly?

Reply:
Its update,
Thanks